File: DecompiledGameScripts/cyberpunk/items/actions/equipAction.swift

public class EquipAction extends BaseItemAction {

  public func CompleteAction(gameInstance: GameInstance) -> Void {
    let obj: wref<GameObject> = this.GetExecutor();
    let req: ref<EquipRequest> = new EquipRequest();
    req.itemID = this.GetItemData().GetID();
    req.owner = obj;
    EquipmentSystem.GetInstance(obj).QueueRequest(req);
  }

  public func IsVisible(const context: script_ref<GetActionsContext>, opt objectActionsCallbackController: wref<gameObjectActionsCallbackController>) -> Bool {
    let obj: wref<GameObject> = this.GetExecutor();
    if IsDefined(obj) {
      return EquipmentSystem.GetInstance(obj).GetPlayerData(obj).IsEquippable(this.GetItemData());
    };
    return true;
  }
}

File: Definitions/Game_Definitions/classes/EquipAction.lua
---@meta
---@diagnostic disable

---@class EquipAction : BaseItemAction
EquipAction = {}

---@return EquipAction
function EquipAction.new() return end

---@param props table
---@return EquipAction
function EquipAction.new(props) return end

function EquipAction:CompleteAction() return end

---@param context gameGetActionsContext
---@param objectActionsCallbackController gameObjectActionsCallbackController
---@return Bool
function EquipAction:IsVisible(context, objectActionsCallbackController) return end

